pkg/genericapiserver/config.go

```diff
@@ -343,48 +352,77 @@ func (c Config) New() (*GenericAPIServer, error) {
 	}
 
 	s.installAPI(&c)
-	s.Handler, s.InsecureHandler = s.buildHandlerChains(&c, http.Handler(s.Mux.BaseMux().(*http.ServeMux)))
+
+	// Wire the handler chain:
+	// - requests from InsecureHandler are marked with skipAuth in the context
```

Comment:
In fact, this horrible construction does not come from the two containers, but from the `InsecureHandler` which skips authn/authz.